Map Editor: highlight cursor

x-Crown opened the pull request:
<img width="1280" height="720" alt="mta-screen_2025-09-26_02-34-39" src="https://github.com/user-attachments/assets/a143159c-4d77-4fdd-bdee-7dda4c8394c5" />
This PR introduces an additional hover highlight effect in the MTA:SA Editor. When the player hovers over an object in the map editor:

The object becomes semi-transparent.

A 3D animated bounding box is drawn around it with rainbow cycling colors (HSV → RGB).

The effect resets automatically when the cursor leaves the object

x-Crown committed:
Update main.lua

This PR introduces an additional hover highlight effect in the MTA:SA Editor.
When the player hovers over an object in the map editor:

The object becomes semi-transparent.

A 3D animated bounding box is drawn around it with rainbow cycling colors (HSV → RGB).

The effect resets automatically when the cursor leaves the object

q8X commented:
i was going to implement this when this PR came out, but it seems like one of the recent changes to editor broke settings (kinda)

https://github.com/multitheftauto/mtasa-resources/blob/604c05e9f1d3e72fd1fad5757b008eefa6002511/%5Beditor%5D/editor_gui/client/options_backend.lua#L133-L137

this issue could delete new settings if the mapper switches to outdated editor (which is likely going to happen when mappers switch between servers)

ArranTuna committed:
Merge branch 'master' into pr/689

ArranTuna committed:
Remove alpha changes because it conflicts.

It doesn't take into account the objects alpha before or after hovering over it, also the properties says the alpha is 150 of any object you select.

ArranTuna committed:
Remove unneeded function

ArranTuna committed:
Add it as a setting disabled by default

ArranTuna committed:
Make it work with all elements

ArranTuna committed:
Move it to a more appropriate file
